src/Microsoft.AspNet.Mvc.Core/ModelBinding/SimpleTypeModelBinder.cs

```diff
@@ -21,38 +19,48 @@ public sealed class TypeConverterModelBinder : IModelBinder
             }
 
             var valueProviderResult = await bindingContext.ValueProvider.GetValueAsync(bindingContext.ModelName);
-            if (valueProviderResult == null)
+            if (valueProviderResult == ValueProviderResult.None)
             {
                 // no entry
                 return null;
             }
 
-            object newModel;
             bindingContext.ModelState.SetModelValue(bindingContext.ModelName, valueProviderResult);
+
             try
             {
-                newModel = valueProviderResult.ConvertTo(bindingContext.ModelType);
-                ModelBindingHelper.ReplaceEmptyStringWithNull(bindingContext.ModelMetadata, ref newModel);
+                var model = valueProviderResult.ConvertTo(bindingContext.ModelType);
+
+                if (bindingContext.ModelType == typeof(string))
+                {
+                    var modelAsString = model as string;
+                    if (bindingContext.ModelMetadata.ConvertEmptyStringToNull &&
+                        string.IsNullOrWhiteSpace(modelAsString))
+                    {
+                        model = null;
+                    }
+                }
+
                 var isModelSet = true;
 
                 // When converting newModel a null value may indicate a failed conversion for an otherwise required
                 // model (can't set a ValueType to null). This detects if a null model value is acceptable given the
                 // current bindingContext. If not, an error is logged.
-                if (newModel == null && !AllowsNullValue(bindingContext.ModelType))
+                if (model == null && !AllowsNullValue(bindingContext.ModelType))
```

Comment:
How will this case occur given the fact `valueProviderResult.ConvertTo()` doesn't return `null` except for reference types anymore? Have we unintentionally lost this error case?